lib/src/doc/field.rs

```diff
@@ -63,7 +63,7 @@ impl<'a> Document<'a> {
 				}
 				// Check for a TYPE clause
 				if let Some(kind) = &fd.kind {
-					val = val.convert_to(kind).map_err(|e| match e {
+					val = val.coerce_to(kind).map_err(|e| match e {
 						// There was a conversion error
 						Error::ConvertTo {
```

Comment:
I'm no where finished reading the PR but it is unclear why `coerce_to` would return `Error::ConvertTo` as opposed to `Error::CoerceTo`.

Reply:
Ahhh good spot. That's supposed to be `Error::CoerceTo` there...